### DISCUSSION TITLE
Tracker image input size vs. Detector image size

### DISCUSSION THREAD

**Discussion by adrian-soch** [mikel-brostrom/yolo_tracking#602] [Q&A] [answered]
Here Yolo takes a resized/preprocessed image `im`:
https://github.com/mikel-brostrom/Yolov5_StrongSORT_OSNet/blob/eab08d1a7bf4b3c63251e9400fe8fe6df6a1c977/track.py#L137

Here the tracker takes the original image `im0`:
https://github.com/mikel-brostrom/Yolov5_StrongSORT_OSNet/blob/eab08d1a7bf4b3c63251e9400fe8fe6df6a1c977/track.py#L188

Why is this the case?

**Comment by mikel-brostrom** [comment]
The only reason is to start with high resolution inputs for the ReID model which then resizes to 128x256. I will run an experiment to determine the metric difference on MOT17 between starting with full resolution and 1280.

**Comment by mikel-brostrom** [comment, marked as answer]
Here are the results: https://github.com/mikel-brostrom/Yolov5_StrongSORT_OSNet/wiki/Effect-of-passing-full-image-input-vs-1280-re-scaled-to-StrongSORT-on-MOT17

|               | 1280          | Full images   |
| ------------- | ------------- | ------------- |
| IDF1          | 70.199        | 72.399        |
| MOTA          | 63.701        | 63.781        |
| HOTA          | 55.69         | 56.783        |

IDF1 takes the largest hit by far, which makes sense. This behavior would be further accentuated by lowering the resolution even more in `val.py`. The execution time is barely affected by using the full image. This is because the detections are resized to 128x256 by the ReID preprocessing before they are fed to the ReID model.
